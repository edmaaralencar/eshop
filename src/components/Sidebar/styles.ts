import styled from 'styled-components'

export const Wrapper = styled.aside`
  /* background-color: ${({ theme }) => theme.colors.primary.light}; */

  width: 25rem;

  display: flex;
  flex-direction: column;

  @media (max-width: 920px) {
    width: auto;
  }
`

type NavItemProps = {
  active: boolean
}

export const NavItem = styled.a<NavItemProps>`
  padding: 1.6rem;
  display: flex;
  align-items: center;
  gap: 0.8rem;
  cursor: pointer;
  border: none;
  text-decoration: none;

  transition: background-color 0.1s ease;

  background-color: ${({ theme, active }) =>
    active ? theme.colors.primary.dark : '#F5F8FA'};
  /* : darken(0.05, theme.colors.primary.light)}; */

  span {
    color: ${({ theme, active }) =>
      active ? theme.colors.white : theme.colors.title};
    font-size: 1.8rem;
    font-weight: 100;
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.primary.dark};

    svg {
      fill: ${({ theme }) => theme.colors.white};
      -webkit-text-stroke-color: white;
    }

    span {
      color: ${({ theme }) => theme.colors.white};
    }
  }
`
