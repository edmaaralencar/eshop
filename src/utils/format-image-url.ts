export function formatImageUrl(image: string) {
  return `/products/${image}`
}
