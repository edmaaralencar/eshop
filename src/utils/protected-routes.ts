import { GetServerSidePropsContext } from 'next'
// import { unstable_getServerSession } from 'next-auth'
import { getSession } from 'next-auth/react'
// import { options } from 'pages/api/auth/[...nextauth].api'

async function protectedRoutes(context: GetServerSidePropsContext) {
  // const session = await unstable_getServerSession(
  //   context.req,
  //   context.res,
  //   options
  // )

  // if (!session) {
  //   return {
  //     redirect: {
  //       destination: `/sign-in?callbackUrl=${context.resolvedUrl}`,
  //       permanent: false
  //     }
  //   }
  // }

  const session = await getSession(context)

  if (!session) {
    context.res.setHeader(
      'Location',
      `/sign-in?callbackUrl=${context.resolvedUrl}`
    )
    context.res.statusCode = 302
  }

  return session
}

export default protectedRoutes
