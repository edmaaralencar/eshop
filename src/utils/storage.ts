const APP_KEY = '@eshop'

export function getStorageItem(key: string) {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(`${APP_KEY}.${key}`)
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return JSON.parse(data!)
}

export function setStorageItem(key: string, value: any[]) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(`${APP_KEY}.${key}`, data)
}

export function removeStorageItem(key: string) {
  if (typeof window === 'undefined') return

  return window.localStorage.removeItem(`${APP_KEY}.${key}`)
}
