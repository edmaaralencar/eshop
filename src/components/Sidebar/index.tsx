import Link from 'next/link'
import { useRouter } from 'next/router'
import { signOut } from 'next-auth/react'

import { RiAccountCircleLine } from 'react-icons/ri'
import { MdOutlineBorderColor, MdLogout } from 'react-icons/md'

import * as S from './styles'

function Sidebar() {
  const router = useRouter()

  return (
    <S.Wrapper>
      <Link href="/profile" passHref legacyBehavior>
        <S.NavItem active={router.pathname === '/profile'}>
          <RiAccountCircleLine
            color={router.pathname === '/profile' ? '#ffffff' : '#18183F'}
            size={24}
          />
          <span>Minha conta</span>
        </S.NavItem>
      </Link>
      <Link href="/orders" passHref legacyBehavior>
        <S.NavItem active={router.pathname === '/orders'}>
          <MdOutlineBorderColor
            color={router.pathname === '/orders' ? '#ffffff' : '#18183F'}
            size={24}
          />
          <span>Meus pedidos</span>
        </S.NavItem>
      </Link>
      {/* <Link href="/change-password" passHref>
        <S.NavItem active={router.pathname === '/change-password'}>
          <RiAccountCircleLine
            color={
              router.pathname === '/change-password' ? '#ffffff' : '#18183F'
            }
            size={24}
          />
          <span>Trocar senha</span>
        </S.NavItem>
      </Link> */}
      <S.NavItem as="button" onClick={() => signOut()} active={false}>
        <MdLogout color="#18183F" size={24} />
        <span>Sair da plataforma</span>
      </S.NavItem>
    </S.Wrapper>
  )
}

export default Sidebar
