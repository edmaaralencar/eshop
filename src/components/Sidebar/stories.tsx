import { StoryObj, Meta } from '@storybook/react'
import Sidebar from '.'

export default {
  title: 'Sidebar',
  component: Sidebar
} as Meta

export const Default: StoryObj = {}
