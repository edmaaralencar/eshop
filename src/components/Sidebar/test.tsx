import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test-helper'

import Sidebar from '.'

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({ pathname: '/profile' }))
}))

describe('<Sidebar />', () => {
  it('should render the links', () => {
    const { container } = renderWithTheme(<Sidebar />)

    expect(screen.getByText(/minha conta/i).parentElement).toHaveAttribute(
      'href',
      '/profile'
    )

    expect(screen.getByText(/meus pedidos/i).parentElement).toHaveAttribute(
      'href',
      '/orders'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the logout button', () => {
    const { container } = renderWithTheme(<Sidebar />)

    expect(
      screen.getByText(/sair da plataforma/i).parentElement
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
