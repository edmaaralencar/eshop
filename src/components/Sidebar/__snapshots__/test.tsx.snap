// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render the links 1`] = `
.c0 {
  width: 25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  cursor: pointer;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease;
  transition: background-color 0.1s ease;
  background-color: #6C6CFF;
}

.c1 span {
  color: #FFFFFF;
  font-size: 1.8rem;
  font-weight: 100;
}

.c1:hover {
  background-color: #6C6CFF;
}

.c1:hover svg {
  fill: #FFFFFF;
}

.c1:hover span {
  color: #FFFFFF;
}

.c2 {
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  cursor: pointer;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease;
  transition: background-color 0.1s ease;
  background-color: #F5F8FA;
}

.c2 span {
  color: #18183F;
  font-size: 1.8rem;
  font-weight: 100;
}

.c2:hover {
  background-color: #6C6CFF;
}

.c2:hover svg {
  fill: #FFFFFF;
}

.c2:hover span {
  color: #FFFFFF;
}

@media (max-width:920px) {
  .c0 {
    width: auto;
  }
}

<aside
  class="c0"
>
  <a
    class="c1"
    href="/profile"
  >
    <svg
      color="#ffffff"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(255, 255, 255);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-4.987-3.744A7.966 7.966 0 0 0 12 20c1.97 0 3.773-.712 5.167-1.892A6.979 6.979 0 0 0 12.16 16a6.981 6.981 0 0 0-5.147 2.256zM5.616 16.82A8.975 8.975 0 0 1 12.16 14a8.972 8.972 0 0 1 6.362 2.634 8 8 0 1 0-12.906.187zM12 13a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-2a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
        />
      </g>
    </svg>
    <span>
      Minha conta
    </span>
  </a>
  <a
    class="c2"
    href="/orders"
  >
    <svg
      color="#18183F"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(24, 24, 63);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-4.987-3.744A7.966 7.966 0 0 0 12 20c1.97 0 3.773-.712 5.167-1.892A6.979 6.979 0 0 0 12.16 16a6.981 6.981 0 0 0-5.147 2.256zM5.616 16.82A8.975 8.975 0 0 1 12.16 14a8.972 8.972 0 0 1 6.362 2.634 8 8 0 1 0-12.906.187zM12 13a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-2a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
        />
      </g>
    </svg>
    <span>
      Meus pedidos
    </span>
  </a>
  <button
    class="c2"
  >
    <svg
      color="#18183F"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(24, 24, 63);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
      />
      <polyline
        points="16 17 21 12 16 7"
      />
      <line
        x1="21"
        x2="9"
        y1="12"
        y2="12"
      />
    </svg>
    <span>
      Sair da plataforma
    </span>
  </button>
</aside>
`;

exports[`<Sidebar /> should render the logout button 1`] = `
.c0 {
  width: 25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  cursor: pointer;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease;
  transition: background-color 0.1s ease;
  background-color: #6C6CFF;
}

.c1 span {
  color: #FFFFFF;
  font-size: 1.8rem;
  font-weight: 100;
}

.c1:hover {
  background-color: #6C6CFF;
}

.c1:hover svg {
  fill: #FFFFFF;
}

.c1:hover span {
  color: #FFFFFF;
}

.c2 {
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  cursor: pointer;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease;
  transition: background-color 0.1s ease;
  background-color: #F5F8FA;
}

.c2 span {
  color: #18183F;
  font-size: 1.8rem;
  font-weight: 100;
}

.c2:hover {
  background-color: #6C6CFF;
}

.c2:hover svg {
  fill: #FFFFFF;
}

.c2:hover span {
  color: #FFFFFF;
}

@media (max-width:920px) {
  .c0 {
    width: auto;
  }
}

<aside
  class="c0"
>
  <a
    class="c1"
    href="/profile"
  >
    <svg
      color="#ffffff"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(255, 255, 255);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-4.987-3.744A7.966 7.966 0 0 0 12 20c1.97 0 3.773-.712 5.167-1.892A6.979 6.979 0 0 0 12.16 16a6.981 6.981 0 0 0-5.147 2.256zM5.616 16.82A8.975 8.975 0 0 1 12.16 14a8.972 8.972 0 0 1 6.362 2.634 8 8 0 1 0-12.906.187zM12 13a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-2a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
        />
      </g>
    </svg>
    <span>
      Minha conta
    </span>
  </a>
  <a
    class="c2"
    href="/orders"
  >
    <svg
      color="#18183F"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(24, 24, 63);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-4.987-3.744A7.966 7.966 0 0 0 12 20c1.97 0 3.773-.712 5.167-1.892A6.979 6.979 0 0 0 12.16 16a6.981 6.981 0 0 0-5.147 2.256zM5.616 16.82A8.975 8.975 0 0 1 12.16 14a8.972 8.972 0 0 1 6.362 2.634 8 8 0 1 0-12.906.187zM12 13a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-2a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
        />
      </g>
    </svg>
    <span>
      Meus pedidos
    </span>
  </a>
  <button
    class="c2"
  >
    <svg
      color="#18183F"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(24, 24, 63);"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
      />
      <polyline
        points="16 17 21 12 16 7"
      />
      <line
        x1="21"
        x2="9"
        y1="12"
        y2="12"
      />
    </svg>
    <span>
      Sair da plataforma
    </span>
  </button>
</aside>
`;
